content/material/2020-01-17-intermediate-dplyr.Rmd

```diff
@@ -0,0 +1,389 @@
+  pivot_longer(-year, names_to = "country_name", values_to = "population_data")
+```
+
+
+**Notice that** in the function `pivot_longer()` the arguments `names_to` and `values_to` require "" because we are providing names of variables that need to be created. 
+Instead, in the function `pivot_wider()` we are giving the arguments `names_from` and `values_from` variables that already exist, hence  the "" are not needed.
+
+#### Exercise
+
+We will use again the `us_rent_income` dataframe for this exercise.
+In this case we want the values of `estimate` and `moe` to be stacked in one single column called `value`, 
+while the respective categories of `estimate` and `moe` should be described in a variable called `calculation`
+
+```{r, eval = FALSE}
+us_rent_income_long <- us_rent_income %>% 
+  #this variable is a nuisance and is not giving information useful to us, so we remove it 
+  select(-GEOID) %>% 
+  #remember that the variable that will not be reshaped need to be omitted using the operator -
+  #when there is more than one variable to omit, we combined the operators - with c()
+  pivot_longer(-c(______, ______), names_to = ________, values_to = _________)
+```
+
+
+
+### Use `pivot_**` functions in combination with `group_by()` and `summarize()` functions
+
+For this example we are going to use the built-in dataframe `storms`
+
+Once again, explore the data using the function `View()`
+
+```{r}
+View(storms)
+```
+
+In this dataframe, information about storms are classified  in long format. 
+Storms are categorized across years and information on wind, latitude, longitude and pressure 
+are presented as columns.
+
+Let's say that we want to see how mean wind speed vary across 
+different storms (name and status) and across years. 
+
+This means that we want to apply the function `pivot_longer()` to visualize each storm name as a separate variable.
+
+However, before reshaping the dataframe in long form, we need to do some extra work. 
+Take another look at the `storms` dataframe. Variables are measured multiple times within each year. 
+So if we want to have a mean value of wind speed for each year, we need to use the functions `group_by()` and `summarize()` that we showed in the intro session. 
+
+We are going to call this new dataframe `storm_wide`
+
+First we will calculate the mean value of wind speed for each storm, across years 
+and discriminating across storm status
+
+```{r}
+storms_wide <- storms %>% 
+  #first we select the variable of interest to avoid an overcrowded dataframe
+  select(name, year, wind, status) %>% 
+  #second we tell R according to which criteria we want to summarize our data, i.e. we want to know the mean value of wind speed for each storm according to storm status and year
+  group_by(name, year, status) %>% 
+  summarize(mean_wind = mean(wind)) 
+```
+
+Now we can reshape the data into a wide format 
+```{r}
+storms_wide <- storms_wide %>% 
+   pivot_wider(names_from = name, values_from = mean_wind)  
+```
```

Comment:
This example is a bit unintuitive, as the resulting wide table is very large, and contains mostly NA (for every storm that was not present in a given year).